plugins/inputs/ceph/ceph.go

```diff
@@ -73,7 +76,22 @@ func (c *Ceph) gatherAdminSocketStats(acc telegraf.Accumulator) error {
 		return fmt.Errorf("failed to find sockets at path %q: %w", c.SocketDir, err)
 	}
 
+	if c.SchemaMaps == nil {
+		c.SchemaMaps = make(map[socket]perfSchemaMap)
+	}
+
 	for _, s := range sockets {
+		if _, ok := c.SchemaMaps[*s]; !ok {
+			rawSchema, err := perfSchema(c.CephBinary, s)
+			if err != nil {
+				c.Log.Warnf("failed to dump perf schema from socket %q: %v", s.socket, err)
+			}
+			if schema, err := parseSchema(rawSchema); err != nil {
+				c.Log.Warnf("failed to parse perf schema from socket %q: %v", s.socket, err)
+			} else {
+				c.SchemaMaps[*s] = schema
+			}
```

Comment:
Do you really mean to continue without a valid `rawSchema`? I don't think this makes sense, does it?

Reply:
Updated logic to only add parsed schema to the schema maps. I added some logic for backward compatibility if schema doesn't exist.